src/Http/Http.Results/src/PhysicalFileResult.cs

```diff
@@ -105,6 +105,14 @@ public Task ExecuteAsync(HttpContext httpContext)
         public FileInfoWrapper(string path)
         {
             var fileInfo = new FileInfo(path);
+
+            // It means we are dealing with a symlink and need to get the information
+            // from the target file instead.
+            if (fileInfo.Exists && !string.IsNullOrEmpty(fileInfo.LinkTarget))
+            {
+                fileInfo = fileInfo.ResolveLinkTarget(true) as FileInfo ?? fileInfo;
```

Comment:
```suggestion
                fileInfo = (FileInfo)fileInfo.ResolveLinkTarget(returnFinalTarget: true) ?? fileInfo;
```
I would just do a direct cast instead of an as cast. This way it doesn't fail silently if the symlink is set up to point to a directory or something strange like that.

Reply:
I am ok with that; however, it will not catch what you are looking because the `ResolveLinkTarget` will always return a `FileInfo` when we call it from a `FileInfo` instance.

https://github.com/dotnet/runtime/blob/c49c65df805304907185bbb4dab909672103ffef/src/libraries/System.Private.CoreLib/src/System/IO/FileSystemInfo.cs#L165

https://github.com/dotnet/runtime/blob/c49c65df805304907185bbb4dab909672103ffef/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs#L469

https://github.com/dotnet/runtime/blob/c49c65df805304907185bbb4dab909672103ffef/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs#L662

Reply:
Just to let you know, it will fail anyway when we try to access the `Length`.